Tasks/JavaToolInstaller/AzureStorageArtifacts/AzureStorageArtifactDownloader.ts

```diff
@@ -0,0 +1,89 @@
+import * as Q from 'q';
+import * as  tl from 'vsts-task-lib/task';
+import msRestAzure = require("azure-arm-rest/azure-arm-common");
+import Model = require("azure-arm-rest/azureModels");
+import armStorage = require('azure-arm-rest/azure-arm-storage');
+import BlobService = require('azure-blobstorage-artifactProvider/blobservice');
+
+export class AzureStorageArtifactDownloader {
+  public connectedService: string;
+  public azureStorageAccountName: string;
+  public containerName: string;
+  public commonVirtualPath: string;
+
+
+  constructor() {
+    this.connectedService = tl.getInput('ConnectedServiceNameARM', true);
```

Comment:
I realize that you probably copied this from the other task, but I don't like the idea of sub-modules doing their own `getInput`.  I'd prefer to see all the `getInput` done in the main script and have their values passed in to these sub-classes.

Reply:
Noted. I'm going to try and extract the azure file out to commons so I won't change this one, but I'll update the other sub-module.